### DISCUSSION TITLE
Type inference debugging

### DISCUSSION THREAD

**Discussion by HolgerGottChristensen** [intellij-rust/intellij-rust#8607] [Q&A]
I am a happy user of intellij-rust and have been for a while. I use it for multiple projects both at work and personal projects.

At work I have a medium sized project where the type inference fails in very simple cases. I have noticed it getting worse over time but am not sure why. I have a personal project the same size that has no problems. 

Below I have included an example of the type inference breaking:
![image](https://user-images.githubusercontent.com/11473146/155710060-d3abc442-f245-42d3-b983-73577205f46a.png)
and 
![image](https://user-images.githubusercontent.com/11473146/155710144-5f9522fa-1e9e-4041-a984-941c5aac4bbb.png)

I want to debug the problems I have and possibly create a more detailed issue, but I dont know where to start and I can't share the specifics of the code.

However some overall things:
* We are using some proc macros and I have turned experimental macro support on. Turning it off did not change the type inference.
* The project has around 300 dependencies and consist of around 6 local crates. The problem exist in all the crates.
* Trying to clear caches and to recreate the project, the problems still occur.

Maybe someone can help getting started with debugging these issues?

Maybe this should be an issue instead but I dont feel I have enough information yet. We can move it if that works better for you.

If you have any questions I will try to provide the information I can within the limits of my employer.

**Comment by LastExceed** [comment]
i have this problem A LOT. unfortunately i can't find a clear pattern to when it happens, which is the bare minimum we need to open an issue. if you have a minimal example that can be reproduced, that would be great

if i copy your example 1:1 everything is working fine, but i know the problem is not on your end because i too have experienced this in the most simple scenarios.

this problem is very frustrating as i simply cannot work without autocomplete and all that good stuff, and i am honestly baffled that this isn't talked about more

**Reply by HolgerGottChristensen** [reply]
This is also why I started this discussion, because I can't find a minimal example either, and it doesnt fail for all projects. 

I want to start debugging the issues if someone has any insight into the type inference in this plugin, but since I have no experience myself the only way I can think of is commenting things out and hope for the best. I am afraid this will take quite a while to locate things like this if I dont really know what I am looking for.

If I find the time I can probably look at just commenting out stuff and maybe at some point have a minimum example.

**Comment by HolgerGottChristensen** [comment]
Clearing all caches and removing all the intellij generated stuff in C:\Users\USERNAME\AppData\Local\Temp seemed to have fixed the issue for now. I also updated to the latest version of the plugin 0.4.164. Hopefully this will help others.
